### ISSUE TITLE
Is there a function of merging two object

### ISSUE THREAD

**Issue by lost22git** [PhilippMDoerner/mapster#7]
someting like :

```nim
type A = object
  name: string
 
type B = object
  name: string

proc merge(a: var A, b: B): discard
```

**Comment by PhilippMDoerner** [comment]
Not so far as the package was conceived on functional grounds to `map` (aka convert) an instance of A to an instance of B.

I have a rough idea on what your reasons might be, but could you type out explicitly why you'd want to merge instead of map?

**Comment by lost22git** [comment]
1. edit entities and save them to db for web backend app

EntityEdit -> Find Entiry by EntityEdit.id -> merge EntityEdit into Entity -> save Entity to db 

2. merge config items for Config system

**Comment by lost22git** [comment]
Similar functionality from the java world 

https://mapstruct.org/documentation/stable/reference/html/#updating-bean-instances

**Comment by PhilippMDoerner** [comment]
Looks like an interesting thing to have. I'll see that I get the testing finished. It'll have to be called `inplaceMap` as `merge` as a keyword is taken by nim.

https://github.com/nim-lang/Nim/issues/22676

**Comment by PhilippMDoerner** [comment]
Feature has been added and is available in the latest version of mapster with v1.1.0.
[There](https://github.com/PhilippMDoerner/mapster/blob/a4397bb2d3a8c366735bdb36e493d3b5eebe145d/src/mapster/inplaceMap.nim#L98) it is called `inplaceMap` as well as `inplaceMapExcept`.

I have not written any docs for it yet, but given that this thing is becoming feature rich enough it might be valid to start a nimibook for it anyway.
